SubterfugeCore/Core/GameEvents/FactoryProduceDrillersEvent.cs

```diff
@@ -83,11 +83,22 @@ public override bool ForwardAction()
         {
             if (Validator.ValidateOutpost(_outpost))
             {
-                if(this._outpost.GetOwner() != null && this._outpost.GetOutpostType() == OutpostType.Factory)
+                if(this._outpost.GetOwner() != null && this._outpost.GetOutpostType() == OutpostType.Factory &&
+                    this._outpost.GetOwner().GetDrillerCount() < this._outpost.GetOwner().GetDrillerCapacity())
                 {
                     this._nextEvent = new FactoryProduceDrillersEvent(this._outpost, this._productionTick.Advance(40));
                     Game.TimeMachine.AddEvent(this._nextEvent);
-                    _outpost.AddDrillers(6);
+                    int drillerDifference = this._outpost.GetOwner().GetDrillerCapacity() - this._outpost.GetOwner().GetDrillerCount();
+
+                    // If capacity would be exceeded, produce only as much drillers as possible without exceeding capacity
+                    if (drillerDifference >= 6)
```

Comment:
We should maybe make the `BASE_FACTORY_PRODUCTION` a constant so that we can easily change it instead of hard setting it to `6`

Reply:
Yup, that's a good idea. Also let's make a BASE_DRILLER_CAPACITY and a BASE_GENERATOR_CAPACITY.
Maybe with a better name.

Reply:
Yeah I like those ideas as well.

Maybe create a global config management class where we can set and load all of the in-game constants. Have an enum of all the constant names and then we can do like `Globals.get(Global.MY_GLOBAL_VALUE)`

Reply:
Will do. Where should I put that class? Core folder or a new folder?

Reply:
Yeah, lets do a new folder, like `Core/Config/` and that can store all the config values